src/parser/GenericParser.ts

```diff
@@ -38,40 +45,67 @@ export class GenericParser {
         let couplingMetrics = {} as CouplingResult;
 
         try {
-            const parseFilesGenerator = findFilesAsync(this.config);
+            const filePathGenerator = findFilesAsync(this.config);
```

Comment:
I think we should add unit tests to be sure that this code behaves as expected on success, on errors, edgecaes etc.

Reply:
For the findFilesAsync-function? I can do that.
For GenericParser? Well, all our current test cases originated from your unit tests for `GenericParser.calculate`, that function is basically our main function that initializes and starts everything related to calculating metrics. I could add some more test cases with files that might trigger errors?

Reply:
Implemented Unit-tests for `GenericParser.calculate` by mocking all more complex called functions within it. 
See `src/parser/GenericParser.test.ts`.